### PR TITLE
Consolidate terraform definitions into one, use common scripts

### PR DESCRIPTION
**Pull request by Chuxel**
@cmendible This PR consolidates the two terraform containers into one with a build arg. It also uses some common scripts instead of having the contents inline in the Dockerfile and uses option args to let you pick what you want to install.

Look good to you?

### CODE REVIEW

**Comment by cmendible** [comment]
@Chuxel let me run a couple of tests and come back to you with the results.

File: containers/azure-terraform/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -1,7 +1,16 @@
 {
-	"name": "Azure Terraform 0.12",
-	"dockerFile": "Dockerfile",
-	"mounts": [ "source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind" ],
+	"name": "Azure Terraform",
+	"build": {
+		"dockerfile": "Dockerfile",
+		"args": { 
+			"TERRAFORM_VERSION": "0.12.16",
```
Comment (cmendible) [inline containers/azure-terraform/.devcontainer/devcontainer.json:6]: We should update to the latest version **0.12.29**

Comment (Chuxel) [reply containers/azure-terraform/.devcontainer/devcontainer.json:6]: Sounds good.  Also bumped up tflint to latest (`0.18.0`).

**Comment by Chuxel** [comment]
> @Chuxel let me run a couple of tests and come back to you with the results.

Sounds good. Bumped up the tf / tflint version.  Let me know if it looks good to merge.

**Comment by cmendible** [comment]
LGTM

**Pull request merged**